TACC/Core-Portal
# task/TV3-161: Shared Workspaces Push Keys

## Overview
- show push keys modal for shared workspaces systems
- clarify that editing permissions or descriptions/titles on systems requires `owner` role
- use `readOnly` flag in a projects-scheme object to render view/manage team
- show roles always in the "view team" modal


## Related

* [TV3-161](https://jira.tacc.utexas.edu/browse/TV3-161)

## Testing

1. Go to https://cep.test/workbench/data/tapis/projects/cep.project.CEPV3-DEV-345
2. Confirm push keys prompt shows up
3. Push keys and confirm you have access (after a few minutes)
4. Set `readOnly` to `true` in the shared workspaces object in `_PORTAL_DATAFILES_STORAGE_SYSTEMS`
5. View your shared workspaces and confirm that you can only view teams, but can edit descriptions where you should be able to

## UI

"View Team" when `readOnly` is `true`
![Screen Shot 2023-04-13 at 4 20 33 PM](https://user-images.githubusercontent.com/20326896/231887670-7cc01214-4787-4f06-9347-e8389b982a41.png)

"View Team" when `readOnly` is `false`, but the user does not have an `owner` or `edit` role:
![Screen Shot 2023-04-13 at 4 20 47 PM](https://user-images.githubusercontent.com/20326896/231887675-08794c5f-7b22-4018-9acf-e2c716423074.png)
